packages/amazon-pay-integration/src/amazon-pay-v2-button-strategy.ts

```diff
@@ -0,0 +1,145 @@
+import {
+    AmazonPayV2CheckoutSessionConfig,
+    AmazonPayV2InitializeOptions,
+    AmazonPayV2PaymentProcessor,
+    AmazonPayV2PayOptions,
+    AmazonPayV2Placement,
+} from '@bigcommerce/checkout-sdk/amazon-pay-utils';
+import {
+    BuyNowCartCreationError,
+    CheckoutButtonInitializeOptions,
+    CheckoutButtonStrategy,
+    getShippableItemsCount,
+    InvalidArgumentError,
+    MissingDataError,
+    MissingDataErrorType,
+    PaymentIntegrationService,
+} from '@bigcommerce/checkout-sdk/payment-integration-api';
+
+import {
+    isWithBuyNowFeatures,
+    WithAmazonPayV2ButtonInitializeOptions,
+    WithBuyNowFeature,
+} from './amazon-pay-v2-button-options';
+import AmazonPayV2RequestSender from './amazon-pay-v2-request-sender';
+import AmazonPayV2ConfigCreationError from './errors/amazon-pay-v2-config-creation-error';
+
+export default class AmazonPayV2ButtonStrategy implements CheckoutButtonStrategy {
+    private _buyNowInitializeOptions: WithBuyNowFeature['buyNowInitializeOptions'];
+
+    constructor(
+        private paymentIntegrationService: PaymentIntegrationService,
+        private amazonPayV2PaymentProcessor: AmazonPayV2PaymentProcessor,
+        private amazonPayV2ConfigRequestSender: AmazonPayV2RequestSender,
+    ) {}
+
+    async initialize(
+        options: CheckoutButtonInitializeOptions & WithAmazonPayV2ButtonInitializeOptions,
+    ): Promise<void> {
+        const { methodId, containerId, amazonpay } = options;
+        const { buttonColor } = amazonpay || {};
+
+        if (!methodId || !containerId) {
+            throw new InvalidArgumentError(
+                'Unable to proceed because "methodId" or "containerId" argument is not provided.',
+            );
+        }
+
+        const { getPaymentMethodOrThrow } = this.paymentIntegrationService.getState();
+
+        const paymentMethod = getPaymentMethodOrThrow<AmazonPayV2InitializeOptions>(methodId);
+        const { initializationData } = paymentMethod;
+
+        await this.amazonPayV2PaymentProcessor.initialize(paymentMethod);
+
+        if (!amazonpay) {
+            await this.paymentIntegrationService.loadDefaultCheckout();
+        }
```

Comment:
Correct me if I am wrong. As I understand `loadDefaultCheckout` should be called only for BuyNow flow, right? So in this case it might be better to stick to `!amazonpay || !isWithBuyNowFeatures(amazonpay)` or something like that.

I am asking since we don't know in what cases we can have `amazonpay` as `undefined`